plugins/plextor/qscan_cmd_plextor.cpp

```diff
@@ -0,0 +1,795 @@
+        peaks[k][j1] = i;
+        local_max = src[k][i];
+        next_peak = 1;
+      } else if (peak_found)
+        if (/*src[k][i-3] >= src[k][i-1] && src[k][i-2] > src[k][i-1] && */
+            src[k][i - 1] > src[k][i] && src[k][i + 1] >= src[k][i]) {
+          mins[k][j2] = i;
+          if (j2 < 13)
+            j2++;
+          peak_found = 0;
+        }
+
+      if (local_max > 2 * src[k][i]) {
+        local_max = 2 * src[k][i];
+        if (next_peak) {
+          next_peak = 0;
+          if (j1 < 13) {
+            j1++;
+            peak_found = 1;
+            //	printf("%4d",i);
+          }
+        }
+      }
+    }
+
+    int min_count = j2;
+    for (i = 0; i < min_count; i++) {
+      int start = (i ? mins[k][i - 1] : 0);
+      int end = mins[k][i];
+      int sum = 0;
+      int partsum = 0;
+      int j;
+      for (j = start; j < end; sum += src[k][j++])
+        ;
+      for (j = start; partsum < sum / 2; partsum += src[k][j++])
+        ;
+      peaks[k][i] = (peaks[k][i] + j - 1) / 2;
+    }
+  }
+  return 0;
+}
+
+int scan_plextor::cmd_dvd_ta_block(cdvd_ta *data) {
+  //	printf("Start PLEXTOR TA test on %d layers\n",dev->media.layers);
+  int i, r, m, j;
+  float sum;
+  unsigned char scan_cmd[6][2] = {{0x04, 0x00}, {0x10, 0x00}, {0x20, 0x00},
+                                  {0xFA, 0x28}, {0xEA, 0x28}, {0xDE, 0x28}};
+  const char *scan_txt[] = {
+      "Running TA on L0 inner zone ", "Running TA on L0 middle zone",
+      "Running TA on L0 outer zone",  "Running TA on L1 inner zone ",
+      "Running TA on L1 middle zone", "Running TA on L1 outer zone"};
+  //	int ta_response_pit[6][512]; int ta_response_land[6][512];
+  int peaks_lands[15], peaks_pits[15];
+  int mins_lands[15], mins_pits[15];
+  int *peaks[] = {peaks_pits, peaks_lands};
+  int *mins[] = {mins_pits + 1, mins_lands + 1};
+
+  if (data->pass < 0 || data->pass > 6)
+    return -1;
+
+  wait_unit_ready(dev, 6);
+  //	for (int pass=0;(pass<3*dev->media.layers) && (!ctl->skip());pass++) {
+  printf("%s\n", scan_txt[data->pass]);
+
+  data->clear();
+  for (i = 0; i < 9; i++) {
+    dev->cmd[0] = PLEXTOR_SCAN_TA_FETE; // 0xF3;
+    dev->cmd[1] = 0x1F;
+    dev->cmd[2] = 0x23;
+    dev->cmd[3] = 0x00;
+    dev->cmd[4] = 0x00;
+    dev->cmd[5] = scan_cmd[data->pass][0];
+    dev->cmd[6] = scan_cmd[data->pass][1];
+    dev->cmd[7] = i << 4;
+    dev->cmd[8] = 0xFF;
+    dev->cmd[9] = 0xFE;
+    dev->cmd[10] = 0x04 * !i;
+    dev->cmd[11] = 0x00;
+    dev->cmd.transport(READ, dev->rd_buf, 65534);
+    printf(".\n");
+    if (!strncmp(dev->dev, "DVDR   PX-714A", 14) ||
+        !strncmp(dev->dev, "DVDR   PX-716A", 14))
+      build_TA_histogram_px716(dev->rd_buf, data->pit, data->land, 512);
+    else
+      build_TA_histogram_px755(dev->rd_buf, data->pit, data->land, 512,
+                               dev->media.type);
+    // dev->parms.scan_speed_dvd);
+  } //////
+  mins_lands[0] = 0;
+  mins_pits[0] = 0;
```

Comment:
Codacy found an issue: [Variable 'mins_pits[0]' is assigned a value that is never used.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6072101)